www/perl-lib/SMT/Utils.pm

```diff
@@ -127,6 +129,46 @@ sub db_connect
     return $dbh;
 }
 
+=item hasRegSharing()
+
+Check whether registration sharing is enabled or not.
+
+=cut
+
+sub hasRegSharing
+{
+    my $r = shift;
+
+    if (! defined $REGSHARING) {
+        $REGSHARING = 0;
+        my $cfg;
+        eval
+        {
+            $cfg = SMT::Utils::getSMTConfig();
+        };
+        if($@ || !defined $cfg)
+        {
+            if (! $r)
+            {
+                return;
+            }
+            $r->log_error("Cannot read the SMT configuration file: ".$@);
+            my $msg = 'SMT server is missconfigured. Please contact your '
+                . 'administrator.';
+            return http_fail($r, 500, $msg);
```

Comment:
Registration.pm lines 166-171 is inside of the function register().
The result of register() is not checked.

Here in Utils.pm you write a function which should return a (kind of) boolean value - True, False and "undef" on error condition.
But http_fail return a true value and would lead into funny behavior.

I think you could test it easily by force some parsing errors or something like this.